#!/usr/bin/env python
PACKAGE = "velocity_controller"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

#gen.add(name, type, level, description, default, min, max)
gen.add("VctrlMax_Y", double_t, 0, "MaxSpeed of Y m/s", 7.0, 0.1, 20.0)
gen.add("VctrlMax_Yaw", double_t, 0, "MaxSpeed of Yaw rad/s", 0.5, 0.01, 1.57)
gen.add("RotEffect", double_t, 0, "Rotation Effect Coefficient", 10.0, 1.0, 100.0)

gen.add("TdiffMax", double_t, 0, "Max Rate of Thrust Differentiation 1/100%", 0.002, 0.0001, 0.01)

gen.add("PGain_Y", double_t, 0, "P Gain of Y", 2.0, 0.0, 20.0)
gen.add("PGain_Yaw", double_t, 0, "P Gain of Yaw", 8.0, 0.0, 20.0)

gen.add("IGain_Y", double_t, 0, "I Gain of Y", 6.0, 0.0, 20.0)
gen.add("IGain_Yaw", double_t, 0, "I Gain of Yaw", 2.0, 0.0, 20.0)


exit( gen.generate(PACKAGE, "velocity_controller", "ControlParam") )
